### ISSUE TITLE
Issues in Xlib.ext.dpms

### ISSUE THREAD

**Issue by thiagokokada** [python-xlib/python-xlib#170]
As described in PR https://github.com/python-xlib/python-xlib/pull/169, putting it here to not lose track of it.

> - `DPMSInfo` doesn't work. I don't have a clue why, but it doesn't seem to be an issue with X11. I receive the following error when trying to call `dpms_info()`:
>
>  ```
>  $ cat playground.py
>  from Xlib import display
>
>  d = display.Display()
>  print(d.dpms_info())
>
>  $ python3 playground.py
>  Traceback (most recent call last):
>    File "playground.py", line 4, in <module>
>      print(d.dpms_info())
>    File "Xlib/display.py", line 225, in __getattr__
>      return types.MethodType(function, self)
>  TypeError: first argument must be callable
>  ```

I did some initial investigation of the issue but I didn't found the root cause. Need some more debugging, but help is appreciated.

> - Any `rq.Request` call (`dpms_enable()`, `dpms_set_timeouts()`, `dpms_force_level()`) only works if you call another `rq.ReplyRequest` function (`dpms_get_version()`, for example) afterwards. I tried to  change them to use `rq.ReplyRequest`, this worked but since they don't return anything they got stucked waiting for a response from X11. Maybe I am missing some call to another function?

This could easily be fixed by calling some some other function that uses `rq.Request`, for example:

```python
def set_timeouts(self, standby_timeout, suspend_timeout, off_timeout):
    DPMSSetTimeouts(display=self.display,
                           opcode=self.display.get_extension_major(extname),
                           major_version=1,
                           minor_version=1,
                           standby_timeout=standby_timeout,
                           suspend_timeout=suspend_timeout,
                           off_timeout=off_timeout)
    return get_timeouts()
```

But I don't know if this is a good solution or if we just ends up covering for possible X11 protocol issues. @vasily-v-ryabov WDYT :thinking: ?

**Comment by vasily-v-ryabov** [comment]
I'm not familiar enough with X11 protocol. No ideas, sorry. I'm maintaining this package as it is used as a dependency for another library (my main hobby project). That means rare releases and code review only. I can pack next release if you feel this is enough for your needs so far. If you're thinking about few more improvements, please let me know when it's done.

**Comment by thiagokokada** [comment]
This is completely fair, thank you @vasily-v-ryabov.

I want to try to at least make `DPMSInfo` work before a release, because without it is kind difficult to work with DPMS (for example, you can't know if the DPMS is actually enabled or disabled and which power level your monitor currently is).

The other issue is workaroundable but it makes for a ugly API :sweat_smile: .

**Comment by petli** [comment]
The X protocol is wholly asynchronous, which is partially hidden by the API (which closely follows the old C Xlib, but today would rather have been implemented as a proper async python API).

This means that it doesn't necessarily send anything until you wait for a server response, either by issuing a request that has a response, or waiting for an event. Small test programs that just send some requests that doesn't have any responses can run into this. For this purpose there is the `sync()` method on the `Display` (which actually does send a small dummy request that returns a response).

**Comment by thiagokokada** [comment]
Closed by #171 .
